.github/workflows/deploy-to-k8s.yaml

```diff
@@ -0,0 +1,68 @@
+name: deploy-to-k8s
+on:
+  workflow_call:
+    inputs:
+      KUBECTL_VERSION:
+        description: 'kubectl version'
+        type: string
+        required: true
+      NAMESPACE:
+        description: 'namespace of the k8s cluster'
+        type: string
+        required: true
+      SECRET_FROM_ENV_PATH:
+        description: 'path to the k8s secret from env yaml file'
+        type: string
+      SECRET_NAME:
+        description: 'name for the k8s secret'
+        type: string
+      MANIFEST_OR_FOLDER_PATH:
+        description: 'path to the manifest yaml file or folder'
+        type: string
+        required: true
+    secrets:
+      KUBE_CONFIG_BASE64:
+        description: 'kube config file in base64 encoded'
+        required: true
+
+jobs:
+  deploy-k8s:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout caller repo
+        uses: actions/checkout@v4
+      - name: Setup kubectl
+        uses: azure/setup-kubectl@v3
+        with:
+          version: ${{ inputs.KUBECTL_VERSION }}
+        id: install
+      - name: Get KUBECONFIG
+        id: get_kubeconfig
+        run: |
+          echo "${{ secrets.KUBE_CONFIG_BASE64 }}" | base64 -d >> ${{ github.workspace }}/kubeconfig
+          KUBECONFIG=${{ github.workspace }}/kubeconfig
+          echo "::add-mask::$KUBECONFIG"
+          echo "KUBECONFIG=$KUBECONFIG" >> $GITHUB_OUTPUT
```

Comment:
Is there a reason we're using `$GITHUB_OUTPUT` instead of `$GITHUB_ENV`? It doesn't require us to import the output in each step's `env:` field.

Reply:
Oh yeah, thanks for mentioning, I was doing some experiments and forgot to change it